include/odp/api/spec/traffic_mngr.h

```diff
@@ -1665,6 +1671,20 @@ int odp_tm_queue_disconnect(odp_tm_queue_t tm_queue);
  */
 int odp_tm_enq(odp_tm_queue_t tm_queue, odp_packet_t pkt);
 
+/** The odp_tm_enq_multi() function is used to add packets to a given TM system.
+ * This API takes multiple packets to enqueue unlike odp_tm_enq() but
+ * rest of the behaviour is identical to that of odp_tm_enq().
+ *
+ * @param tm_queue  Specifies the tm_queue (and indirectly the TM system).
+ * @param packets   Array of packets to enqueue.
+ * @param num       Number of packets to send.
+ * @return          Returns >0 upon success indicating number of packets
+ *                  enqueued, <= 0 upon failure. One of the more common
+ *                  failure reasons is WRED drop.
+ */
+int odp_tm_enq_multi(odp_tm_queue_t tm_queue, const odp_packet_t packets[],
+		     int num);
```

Comment:
As commented previously, returning failure because of WRED drop is not going to work well. If the N:th packet would cause WRED drop, then it does not make sense for the function to return failure (indicating that no packets were enqueued). Just consuming the first N-1 packets would not be good either because it would require multiple retries to get a batch of packets processed (enqueued or WRED dropped). Maybe the function should just consume all the packets, not indicating if some of them got WRED dropped.

Reply:
Will remove the test indicating RED drop would return as failure

Reply:
I think this is still slightly problematic.

The description says: "[the] rest of the behaviour is identical to that of odp_tm_enq()." And the text for odp_tm_enq() says "One of the more common failure reasons is WRED drop."

I wonder if odp_tm_enq() should no longer be allowed to return failure in WRED drop case, but if that is not changed, it might be good to say in the description of odp_tm_enq_multi() that WRED drops do not cause failures of the call. Also, "number of packets enqueued" could be changed to "number of packets consumed" so that it does not imply that all the packets ended up in the queue.

Reply:
Ok. I'll change just odp_tm_enq_multi() to say that WRED drops do not cause failures of the call.